src/internal/InnerClickListener.js

```diff
@@ -38,10 +38,12 @@ export default class InnerClickListener extends React.Component {
   }
 
   handleDocumentClick(event) {
-    if (this.element) {
-      if (this.element.contains && !this.element.contains(event.target)) {
-        this.props.onClickOutside(event);
-      }
+    // Ensure that the target exists in the DOM before checking the element
+    // documentElement necessary for IE
+    if (document.documentElement.contains(event.target) && this.element) {
```

Comment:
I believe we avoid usage of globals like `document` or `window` to prevent issues with scoping cc @asudoh

```suggestion
    if (this.element.ownerDocument.contains(event.target) && this.element) {
```